gitpod-io/gitpod
# [prebuilds] increment metric only with state change

## Description
This PR makes sure that metrics are only incremented when a prebuild state changes.
It also extends the logging information with the logging context.

## Related Issue(s)
Fixes #10616

## How to test
1. Trigger multiple prebuilds (does not have to be at the same time) and wait until they are completed.
2. Check that the metrics show up on this page for [`started`](https://9090-gitpodio-gitpod-e2np1izepbl.ws-eu47.gitpod.io/graph?g0.expr=sum(gitpod_prebuilds_started_total)&g0.tab=1&g0.stacked=0&g0.show_exemplars=0&g0.range_input=1h) and [`completed`](https://9090-gitpodio-gitpod-e2np1izepbl.ws-eu47.gitpod.io/graph?g0.expr=sum(gitpod_prebuilds_completed_total)&g0.tab=1&g0.stacked=0&g0.show_exemplars=0&g0.range_input=1h).

## Release Notes

```release-note
NONE
```

## Documentation
